frame/staking-rewards/src/lib.rs

```diff
@@ -258,13 +320,80 @@ pub mod pallet {
 
 		#[transactional]
 		fn stake(
-			_who: &Self::AccountId,
-			_pool_id: &Self::RewardPoolId,
-			_amount: Self::Balance,
-			_duration_preset: DurationSeconds,
-			_keep_alive: bool,
+			who: &Self::AccountId,
+			pool_id: &Self::RewardPoolId,
+			amount: Self::Balance,
```

Comment:
Need to transfer this amount to an account derived based on the pallet and pool id

Reply:
Done.